syntax = "proto3";
package interpreter_IPC.toInterpreter;

import "interpreterInit.proto";


message System {
	enum EventType {
		kRequestStop = 0;
	}
	EventType type = 1;
}

message AvalaibleRequest {
}

message LoadScriptContentRequest {
	string scriptPath = 1;
}

message SaveScriptContentRequest {
	string scriptPath = 1;
	string content = 2;
}

message StartScript {
	int32 scriptInstanceId = 1;
	string scriptPath = 2;
	string scriptApiId = 3;
	string scriptLogPath = 4;
}

message StopScript {
	int32 scriptInstanceId = 1;
}

message PurgeScriptLog {
	int32 scriptInstanceId = 1;
}


message msg {
	oneof OneOf {
		string error = 1;
		Init init = 2;
		System system = 3;
		AvalaibleRequest avalaibleRequest = 4;
		LoadScriptContentRequest loadScriptContentRequest = 5;
		SaveScriptContentRequest saveScriptContentRequest = 6;
		StartScript startScript = 7;
		StopScript stopScript = 8;
		PurgeScriptLog purgeScriptLog = 9;
   }
}

